openmc/plotter.py

```diff
@@ -95,37 +124,62 @@ def plot_xs(this, types, divisor_types=None, temperature=294., axis=None,
 
+    cv.check_type("plot_CE", plot_CE, bool)
+
     if isinstance(this, openmc.Nuclide):
         data_type = 'nuclide'
     elif isinstance(this, openmc.Element):
         data_type = 'element'
     elif isinstance(this, openmc.Material):
         data_type = 'material'
+    elif isinstance(this, openmc.Macroscopic):
+        data_type = 'macroscopic'
     else:
         raise TypeError("Invalid type for plotting")
 
-    E, data = calculate_xs(this, types, temperature, sab_name, cross_sections,
-                           enrichment)
-
-    if divisor_types:
-        cv.check_length('divisor types', divisor_types, len(types),
-                        len(types))
-        Ediv, data_div = calculate_xs(this, divisor_types, temperature,
-                                      sab_name, cross_sections, enrichment)
-
-        # Create a new union grid, interpolate data and data_div on to that
-        # grid, and then do the actual division
-        Enum = E[:]
-        E = np.union1d(Enum, Ediv)
-        data_new = np.zeros((len(types), len(E)))
-
-        for line in range(len(types)):
-            data_new[line, :] = \
-                np.divide(np.interp(E, Enum, data[line, :]),
-                          np.interp(E, Ediv, data_div[line, :]))
-            if divisor_types[line] != 'unity':
-                types[line] = types[line] + ' / ' + divisor_types[line]
-        data = data_new
+    if plot_CE:
+        # Calculate for the CE cross sections
+        E, data = calculate_cexs(this, types, temperature, sab_name,
+                                 ce_cross_sections, enrichment)
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types),
+                            len(types))
+            Ediv, data_div = calculate_cexs(this, divisor_types, temperature,
+                                            sab_name, ce_cross_sections,
+                                            enrichment)
+
+            # Create a new union grid, interpolate data and data_div on to that
+            # grid, and then do the actual division
+            Enum = E[:]
+            E = np.union1d(Enum, Ediv)
+            data_new = np.zeros((len(types), len(E)))
+
+            for line in range(len(types)):
+                data_new[line, :] = \
+                    np.divide(np.interp(E, Enum, data[line, :]),
+                              np.interp(E, Ediv, data_div[line, :]))
+                if divisor_types[line] != 'unity':
+                    types[line] = types[line] + ' / ' + divisor_types[line]
+            data = data_new
+    else:
+        # Calculate for MG cross sections
+        E, data = calculate_mgxs(this, types, orders, temperature,
+                                 mg_cross_sections, ce_cross_sections,
+                                 enrichment)
+
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types),
+                            len(types))
```

Comment:
No need to specify the fourth argument here (it is assumed to be the same if not specified)